bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,450 @@
+* The management inet address is bound to the 0.0.0.0 inet address allowing for local (required by WildFly CLI) and remote access (required by OpenShift readiness and liveness probes).
+* The transaction subsystem id is set to the value of ```jboss.node.name```.
+* The ```jboss.node.name``` system propery, if not set, is set to the value of ```HOSTNAME``` environment variable if defined (defined in OpenShift PODS). The node name value
+is truncated to a max of 23 characters in order for the transaction subsystem to properly operate.
+* The server logs are printed in the console.
+* jgroups subsystem is configured to use kubernetes.KUBE_PING jgroups protocol for both tcp (default stack) and udp. PING and MPING protocols are removed.
+* It is possible to configure jgroups to use un-encrypted password authentication. Set the ```<cloud>``` child element ```<enable-jgroups-password>true|false</enable-jgroups-password>``` to enable authentication. 
+NB: When authentication is enabled, the environment variable ```JGROUPS_CLUSTER_PASSWORD``` must be set.
+
+Some examples:
+
+Configure for OpenShift execution:
+
+```
+<cloud/>
+```
+
+Configure for OpenShift execution with jgroups authentication enabled:
+
+```
+<cloud>
+  <enable-jgroups-password>true</enable-jgroups-password>
+</cloud>
+```
+Configure for kubernetes execution:
+
+```
+<cloud>
+  <type>kubernetes</type>
+</cloud>
+```
+
+===== WildFly OpenShift operator
+
+The WildFly OpenShift operator can be used to manage deployments based on image containing a WildFly bootable JAR.
+At boot time, the WildFly bootable JAR dumps in the file ```/opt/jboss/container/wildfly-bootable-jar/install-dir``` its installation path.
+This information is required by the WildFly OpenShift operator to retrieve transaction logs and call into WildFly CLI.
+
+===== JKube Maven plugin
+
+The JKube Maven Plugin is evolved with a generator that recognizes the bootable JAR maven plugin. DETAILS TO BE PROVIDED WHEN FINALIZED.
+
+=== Nice-to-Have Requirements
+
+* Add the ability to copy content in the server during build. That is a common requirement (e.g.: auth properties files).
+* Ability to generate a runtime Maven repository in order to resolve the server artifacts from Maven local cache. 
+Huge benefit in term of JAR size and boot time (4/5 time faster, around 160ms to start vs 700 ms). 
+Specifically in a docker/openshift context, relying on slim server + Maven repository speeds up startup without impacting image size.
+Eg: java -Dmaven.repo.local=/maven-repo -jar myapp-bootable.jar
+
+
+=== Non-Requirements
+
+* Offer a new way to configure the server (e.g.: Thorntail yaml file).
+* Package a custom standalone XML file. Standalone XML file is generated during build by the Maven plugin and can't be replaced. 
+* Usage of a different log manager than JBoss log manager is out of scope.
+* No domain support.
```

Comment:
Can an item be added here to inform that the Thorntail auto-detection feature replacement is out of scope?

Reply:
Ok, Added.

Reply:
Great, thanks.